src/NavBar.js

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import {
+  AppBar,
+  Button,
+  Toolbar,
+  Typography,
+} from '@material-ui/core';
+import { useAuth } from './AuthProvider';
+
+function NavBar() {
+  const auth = useAuth();
+  const handleSignOut = () => {
+    auth.signOut();
+  };
+  const signOutButton = (
+    auth.user
+      ? <Button onClick={handleSignOut}>Sign Out</Button>
+      : <div />
+  );
+
+  return (
+    <AppBar position="static">
+      <Toolbar>
+        <Typography variant="h6" color="inherit" style={{ flexGrow: 1 }}>
+          Job Portal
+        </Typography>
+        <div>
+          {signOutButton}
+        </div>
```

Comment:
Unless I'm missing something, you don't need extra divs here, you can simply render the button only when logged in:
```
const isLoggedIn = !!auth.user;
//...
<Typography variant="h6" color="inherit" style={{ flexGrow: 1 }}>
  Job Portal
</Typography>
{isLoggedIn && <Button onClick={handleSignOut}>Sign Out</Button>}
```